sound/soc/sof/ipc.c

```diff
@@ -636,6 +636,48 @@ int snd_sof_ipc_stream_posn(struct snd_sof_dev *sdev,
 }
 EXPORT_SYMBOL(snd_sof_ipc_stream_posn);
 
+static int sof_get_ctrl_ipc_dir(enum sof_ipc_ctrl_type ctrl_type)
```

Comment:
I'm not sure this is an improvement... With my comment to your previous patch this switch-case would intervene nicely with the rest of the code. As I proposed, I'd split that function differently. Or do you expect to call this function again from other locations? If so, I'd remove the "send" variable and just do "return 1" or "return 0" from the switch directly.